Packet++/src/TcpReassembly.cpp

```diff
@@ -581,30 +587,30 @@ void TcpReassembly::checkOutOfOrderFragments(TcpReassemblyData* tcpReassemblyDat
 
 		uint32_t closestSequence = 0xffffffff;
 		bool closestSequenceDefined = false;
-		int closestSequenceFragIndex = -1;
-		index = 0;
+		auto closestSequenceFragIt = tcpReassemblyData->twoSides[sideIndex].tcpFragmentList.end();
+		it = tcpReassemblyData->twoSides[sideIndex].tcpFragmentList.begin();
 
-		while (index < (int)tcpReassemblyData->twoSides[sideIndex].tcpFragmentList.size())
+		while (it != tcpReassemblyData->twoSides[sideIndex].tcpFragmentList.end())
 		{
 			// extract segment at current index
-			TcpFragment* curTcpFrag = tcpReassemblyData->twoSides[sideIndex].tcpFragmentList.at(index);
+			TcpFragment* curTcpFrag = *it;
 
 			// check if its sequence is closer than current closest sequence
 			if (!closestSequenceDefined || SEQ_LT(curTcpFrag->sequence, closestSequence))
 			{
 				closestSequence = curTcpFrag->sequence;
-				closestSequenceFragIndex = index;
+				closestSequenceFragIt = it;
 				closestSequenceDefined = true;
 			}
 
-			index++;
+			it++;
 		}
 
 		// this means fragment list is not empty at this stage
-		if (closestSequenceFragIndex > -1)
+		if (closestSequenceFragIt != tcpReassemblyData->twoSides[sideIndex].tcpFragmentList.end())
 		{
 			// get the fragment with the closest sequence
-			TcpFragment* curTcpFrag = tcpReassemblyData->twoSides[sideIndex].tcpFragmentList.at(closestSequenceFragIndex);
+			TcpFragment* curTcpFrag = tcpReassemblyData->twoSides[sideIndex].tcpFragmentList.getAndRemoveFromVector(closestSequenceFragIt);
```

Comment:
ditto: `unique_ptr`

Reply:
Updated in commit https://github.com/seladb/PcapPlusPlus/pull/1346/commits/baac906fd2d26b5abaa2f807af5c8d287db4712c